src/rgw/rgw_auth_keystone.h

```diff
@@ -72,6 +72,57 @@ class TokenEngine : public rgw::auth::Engine {
   }
 }; /* class TokenEngine */
 
+class SecretCache {
+  using token_envelope_t = rgw::keystone::TokenEnvelope;
+
+  struct secret_entry {
+    token_envelope_t token;
+    std::string secret;
+    uint64_t expires;
+    list<std::string>::iterator lru_iter;
+  };
+
+  const boost::intrusive_ptr<CephContext> cct;
+
+  std::map<std::string, secret_entry> secrets;
+  std::list<std::string> secrets_lru;
```

Comment:
it would be preferable to avoid std:list, but all of this is likely fine for now